test/frontend/2dPrintingTest.js

```diff
@@ -0,0 +1,26 @@
+/* global describe it */
+import 'jsdom-global/register'
+import React from 'react'
+import Enzyme, { mount } from 'enzyme'
+import { expect } from 'chai'
+import Printing2D from '../../src/Pages/2DPrinting/App'
+import Adapter from 'enzyme-adapter-react-16'
+import { Jumbotron } from 'reactstrap'
+import { FilePond } from 'react-filepond'
+
+Enzyme.configure({ adapter: new Adapter() })
+
+describe('<Printing2D />', () => {
```

Comment:
Thanks for writing tests. To improve, we can test the values of our `useState` variables as well. This [helpful guide](https://dev.to/theactualgivens/testing-react-hook-state-changes-2oga) explains how to do it. let me know if you'd like to give it a try.

Reply:
I'll give this a try!

Reply:
Cool!